compiler/main.nim

```diff
@@ -266,11 +266,27 @@ proc mainCommand*(graph: ModuleGraph) =
       var libpaths = newJArray()
       for dir in conf.searchPaths: libpaths.elems.add(%dir.string)
 
+      var hints = block: # consider factoring with `listHints`
+        var ret = newJObject()
+        for a in hintMin..hintMax:
+          let key = lineinfos.HintsToStr[ord(a) - ord(hintMin)]
+          ret[key] = % (a in conf.notes)
+        ret
```

Comment:
Dunno why you don't simply use the ``hints`` variable but I'm merging this now and will fix it on my own because this ping-pong is annoying me as much as it annoys you.

Reply:
good point, I saw your fix. And yes, I think it's much more efficient for everyone to merge first and fix later (in many cases); not saying that as an excuse to wrap crappy code in 1st place of course.